service_provider.go

```diff
@@ -363,6 +363,39 @@ func (ivr *InvalidResponseError) Error() string {
 	return fmt.Sprintf("Authentication failed")
 }
 
+func responseContainsSignature(response *etree.Document) (bool, error) {
```

Comment:
This check doesn't work (I think) if the response is encrypted

Reply:
Will fix.